Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,69 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
+    command = CommandsCommand.new
+    command.call
+  end
+
+  class CommandsCommand < Command
+    def initialize
+      super
+      desc "Show a list of built-in and external commands"
+      option "--quiet", "List only the names of commands without the header"
+      option "--include-aliases", "The aliases of internal commands will be included"
+    end
+
+    def call
+      check_invalid_options(ARGV.options_only)
+
+      if ARGV.include? "--quiet"
```

Comment:
This doesn't seem to relate in any way to the options declared in this method (and it should)